### ISSUE TITLE
[no-unused-vars] False positive for unused vars inside decorator arguments

### ISSUE THREAD

**Issue by ridaamirini** [bradzacher/eslint-plugin-typescript#126]
From https://github.com/eslint/typescript-eslint-parser/issues/445

**What code were you trying to parse?**

```
import { Component, Vue } from 'vue-property-decorator';
import HelloWorld from './components/HelloWorld.vue';

@Component({
  components: {
    HelloWorld
  }
})
export default class App extends Vue {}
```

**What did you expect to happen?**

With eslint-plugin-typescript's typescript/no-unused-vars rule enabled, should lint without errors.

**What happened?**

```
error: 'HelloWorld' is defined but never used (no-unused-vars)
```

It looks like the expression in a decorator's argument position is not taken into context when determining unused vars. Also, when using eslint --fix, code inside decorator arguments are sometimes not formatted correctly.

**Comment by Weiyu-Chen** [comment]
So could anyone fix it ?

**Comment by gwardwell** [comment]
I'm seeing this issue as well. Is there any progress on a fix?

**Comment by bradzacher** [comment]
https://astexplorer.net/#/gist/9f8a50b0db8e4f6eb6a4232cb3bf4160/618c7989605019cdd3a3b8d023cb8c6fe47a837f

```
const foo = '';

@deco({
  foo
})
export default class App {}

@deco({
  foo
})
export class App {}

```

Looking at the AST and comparing it to the `markDecoratorAsUsed` function, it looks like it was never built to handle decorator arguments.

Having a look, this will be fixed by the new `visitorKeys` config that the parser added in v20.
So this should be fixed by #166
